prow/plugins/label/label_test.go

```diff
@@ -70,20 +79,141 @@ func getFakeRepo(commentBody, commenter string, repoLabels []string) (*fakegithu
 		},
 		Action: "created",
 	}
-	return fakeCli, ice
+
+	ae := assignEvent{
+		action:  ice.Action,
+		body:    ice.Comment.Body,
+		login:   ice.Comment.User.Login,
+		org:     ice.Repo.Owner.Login,
+		repo:    ice.Repo.Name,
+		url:     ice.Comment.HTMLURL,
+		number:  ice.Issue.Number,
+		issue:   ice.Issue,
+		comment: ice.Comment,
+	}
+
+	return fakeCli, ae
+}
+
+func getFakeRepoIssue(commentBody, commenter string, repoLabels []string) (*fakegithub.FakeClient, assignEvent) {
+	fakeCli := &fakegithub.FakeClient{
+		Issues:         make([]github.Issue, 1),
+		IssueComments:  make(map[int][]github.IssueComment),
+		ExistingLabels: repoLabels,
+		OrgMembers:     []string{orgMember},
+	}
+	startingLabels := []github.Label{}
+
+	ie := github.IssueEvent{
+		Repo: github.Repo{
+			Owner: github.User{Login: fakeRepoOrg},
+			Name:  fakeRepoName,
+		},
+		Issue: github.Issue{
+			User:        github.User{Login: "a"},
+			Number:      prNumber,
+			PullRequest: &struct{}{},
+			Body:        commentBody,
+			Labels:      startingLabels,
+		},
+		Action: "created",
+	}
+
+	ae := assignEvent{
+		action: ie.Action,
+		body:   ie.Issue.Body,
+		login:  ie.Issue.User.Login,
+		org:    ie.Repo.Owner.Login,
+		repo:   ie.Repo.Name,
+		url:    ie.Issue.HTMLURL,
+		number: ie.Issue.Number,
+		issue:  ie.Issue,
+		comment: github.IssueComment{
+			Body: ie.Issue.Body,
+			User: ie.Issue.User,
+		},
+	}
+
+	return fakeCli, ae
+}
+
+func getFakeRepoPullRequest(commentBody, commenter string, repoLabels []string) (*fakegithub.FakeClient, assignEvent) {
+	fakeCli := &fakegithub.FakeClient{
+		PullRequests:   make(map[int]*github.PullRequest),
+		IssueComments:  make(map[int][]github.IssueComment),
+		ExistingLabels: repoLabels,
+		OrgMembers:     []string{orgMember},
+	}
+	startingLabel := github.Label{}
+
+	pre := github.PullRequestEvent{
+		PullRequest: github.PullRequest{
+			User:   github.User{Login: commenter},
+			Number: prNumber,
+			Base: github.PullRequestBranch{
+				Repo: github.Repo{
+					Owner: github.User{Login: fakeRepoOrg},
+					Name:  fakeRepoName,
+				},
+			},
+			Body: commentBody,
+			Head: github.PullRequestBranch{
+				Repo: github.Repo{
+					Owner: github.User{Login: fakeRepoOrg},
+					Name:  fakeRepoName,
+				},
+			},
+		},
+		Label:  startingLabel,
+		Number: prNumber,
+		Action: "created",
+	}
+
+	ae := assignEvent{
+		action: pre.Action,
+		body:   pre.PullRequest.Body,
+		login:  pre.PullRequest.User.Login,
+		org:    pre.PullRequest.Base.Repo.Owner.Login,
+		repo:   pre.PullRequest.Base.Repo.Name,
+		url:    pre.PullRequest.HTMLURL,
+		number: pre.Number,
+		comment: github.IssueComment{
+			Body: pre.PullRequest.Body,
+			User: pre.PullRequest.User,
+		},
+	}
+
+	return fakeCli, ae
+}
+
+func examineTestCase(t *testing.T, fakeClient *fakegithub.FakeClient, ae assignEvent, tc testCase) {
```

Comment:
We should have a more descriptive name for this method here I think. `testLabels` or something maybe?